support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.jsx

```
@@ -0,0 +1,206 @@
+        empty: true,
+        error: '',
+        errorEmpty: 'Please enter a CVC number',
+        errorIncomplete: 'Please enter a valid CVC number',
+      },
+      cardErrors: [],
+    };
+  }
+
+  getAllCardErrors = () => ['cardNumber', 'cardExpiry', 'cardCvc'].reduce((cardErrors, field) => {
+    if (this.state[field].error.length > 0) {
+      cardErrors.push({ field: [field], message: this.state[field].error });
+    }
+    return cardErrors;
+  }, [])
+
+  handleCardErrors = () => {
+    // eslint-disable-next-line array-callback-return
+    ['cardNumber', 'cardExpiry', 'cardCvc'].map((field) => {
+      if (this.state[field].empty === true) {
+        this.setState({
+          [field]: {
+            ...this.state[field],
+            error: this.state[field].errorEmpty,
+          },
+        });
+      } else if (!this.state[field].complete) {
+        this.setState({
+          [field]: {
+            ...this.state[field],
+            error: this.state[field].errorIncomplete,
+          },
+        });
+      }
+      this.setState({ cardErrors: this.getAllCardErrors() });
+    });
+  }
+
+  handleChange = (event) => {
+    if (this.state[event.elementType].error) {
+      this.setState({
+        [event.elementType]: {
+          ...this.state[event.elementType],
+          error: '',
+        },
+      });
+    } else {
+      this.setState({
+        [event.elementType]: {
+          ...this.state[event.elementType],
+          complete: event.complete,
+          empty: event.empty,
+        },
+      });
+    }
+  }
+
+  requestStripeToken = (event) => {
+    event.preventDefault();
+    this.props.validateForm();
+    this.handleCardErrors();
+    if (this.props.stripe && this.props.allErrors.length === 0 && this.state.cardErrors.length === 0) {
+      const { stripe } = this.props;
+      stripe.createToken({ type: 'card', name: this.props.name })
+        .then(({ token, error }) => {
+          if (error) {
+            console.log(error);
```

Do we need to report this error to the user?